tools/accuracy_checker/accuracy_checker/config/config_validator.py

```diff
@@ -152,7 +152,11 @@ def validate(self, entry, field_uri=None):
 
     @property
     def type(self):
-        return str
+        def type_def(x=None):
+            if x is None:
+                return str
+            return x
```

Comment:
I don't understand what's going on here. This function returns `str` (a type) if `x` is `None`, but `x` (a value) otherwise. That doesn't seem right. And why is it needed at all? The logic in `utils.py` already handles the case when `x` is `None`.

Reply:
getting type from Field also used for static analisys required configuration for model in DL WorkBench. They want to get field type without providing argument. The BaseField used in cases, where several types applicable (e.g. for mean, we can provide float, list of floats for each channel or select one of precomputed values using strings), we can not know which type will have input values, but during parsing is opportunity to parsing necessary formats from string always exists.

Reply:
I see; however, your code doesn't seem to actually support this use case: if `f` is an instance of `BaseField`, then `f.type` is `type_def`, not `str`.

IMO, a better way to do this would be to have `type` return `None` here, and add special logic for `None` both in `get_parameter_value_from_config` and in DL WB.

Reply:
ok, changes